api/src/helperClasses.js

```diff
@@ -20,17 +20,11 @@ class BigQueryParser {
   constructor(mapping){
     this.mapping = mapping;
   }
-// https://stackoverflow.com/questions/2218999/how-to-remove-all-duplicates-from-an-array-of-objects
 // this will only ever iterate over a max of 1000 rows, so it won't tank endpoint speed
 // like totally reformatting the query to deduplicate all the data will
   deduplicateData (rows) {
-    const dedupe = rows.filter((row, index) => {
-      const _row = JSON.stringify(row);
-      return index === rows.findIndex(obj => {
-        return JSON.stringify(obj) === _row;
-      });
-    });
-    return dedupe;
+    const set = new Set(rows);
```

Comment:
I still don't understand the idea behind this. You want to avoid duplicates within batches but not across batches... why? If you want the actual full return to be deduplicated, this won't work. If you don't need that, what's the point of deduplicating it slightly? And doing it on the client can't be more efficient than doing it big query if you're doing the same thing, so I still struggle to understand the optimization.